pmlc/ast/ast_ops.cc

```diff
@@ -59,24 +59,29 @@ struct IndexOp : Intrinsic {
 struct GatherOp : Intrinsic {
   TensorShapes getShapes(Evaluator *evaluator, ArrayRef<ExprNodePtr> operands,
                          ArrayRef<TensorShape> shapes) const final {
-    if (operands.size() != 2) {
-      throw std::runtime_error("'gather' requires 2 arguments.");
+    auto operands_size = operands.size();
+    if (operands_size < 2 || operands_size > 5) {
+      throw std::runtime_error("'gather' requires 2-5 arguments.");
```

Comment:
I'd prefer that the `GatherOp` at the ast level always expects a fixed number of operands. This means that the `edsl.h` should add default values if we don't want to force the user to specify all 5 operands in every call.

Reply:
Changed to fixed number (5) of operands for gather.  Added default values to edsl c++ API, but also need to add default values to edsl python API (to support keras backend tests). I wonder if we should use a new function`plaidml_expr_gather` (like`plaidml_expr_pragma`) instead of `intrinsicCall` (which calls `plaidml_expr_intrinsic`).